feature_engine/encoding/decision_tree.py

```diff
@@ -163,6 +169,15 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
             ordered ordinal encoding.
         """
 
+        # confirm model type and target variables are compatible.
+        if self.regression is True and type_of_target(y) == "binary":
+            raise ValueError(f"'regression' is {self.regression} and target "
+                             f"variables are a binary. These two parameter settings "
+                             f"are not compatible.")
+
+        if self.regression is False:
```

Comment:
instead of the if condition I would use `else` here as part of the former if.

Reply:
Ok, I also made a slight revision to the prior `if` statement, so it's more intuitive to the user.